### PR TITLE
Legacy SDK compliance continue

### PR DESCRIPTION
**Pull request by heyams**
Continue to address critical and high CC issues [here](https://github-private.visualstudio.com/microsoft/_componentGovernance/161106?_a=alerts&typeId=6520417&alerts-view-option=active).

### CODE REVIEW

File: logging/log4j1_2/build.gradle [hunk 1]
```diff
@@ -33,5 +33,5 @@ projectPomDescription = "This module provides a $project.msftAppInsights appende
 // endregion Publishing properties
 
 dependencies {
-    compile group: 'log4j', name: 'log4j', version: '1.2.17'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.15.0'
```
Comment (trask) [inline logging/log4j1_2/build.gradle:36]: this is specifically a log4j 1.2 appender, so need to still need to compile against 1.2.x, we can make it compileOnly similar to what we did for log4j2 appender in #2002

File: test/smoke/testApps/TraceLog4j1_2/build.gradle [hunk 2]
```diff
@@ -12,7 +12,7 @@ dependencies {
 
     testCompile 'com.google.guava:guava:23.0' // VSCODE intellisense bug workaround
 
-    compile group: 'log4j', name: 'log4j', version: '1.2.17'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.15.0'
```
Comment (trask) [inline test/smoke/testApps/TraceLog4j1_2/build.gradle:15]: this smoke test is specifically testing log4j 1.2

File: test/smoke/testApps/TraceLog4j1_2UsingAgent/build.gradle [hunk 3]
```diff
@@ -10,7 +10,7 @@ dependencies {
 
     testCompile 'com.google.guava:guava:23.0' // VSCODE intellisense bug workaround
 
-    compile group: 'log4j', name: 'log4j', version: '1.2.17'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.15.0'
```
Comment (trask) [inline test/smoke/testApps/TraceLog4j1_2UsingAgent/build.gradle:13]: same here

Comment (heyams) [reply test/smoke/testApps/TraceLog4j1_2UsingAgent/build.gradle:13]: good catch. i will revert tests.. i did a global replace.

File: logging/log4j1_2/build.gradle (hunk 1 above)
Comment (heyams) [reply logging/log4j1_2/build.gradle]: Update.

File: logging/log4j1_2/build.gradle [hunk 4]
```diff
@@ -33,5 +33,5 @@ projectPomDescription = "This module provides a $project.msftAppInsights appende
 // endregion Publishing properties
 
 dependencies {
-    compile group: 'log4j', name: 'log4j', version: '1.2.17'
+    compileOnly group: 'log4j', name: 'log4j', version: '1.2.17'
 }
```
Comment (trask) [inline logging/log4j1_2/build.gradle:37]: if the tests fail, that probably means you need to pull in log4j as testImplementation (see #2002)

**Review by trask — APPROVED** [review]

File: logging/log4j1_2/build.gradle [hunk 5]
```diff
@@ -33,5 +33,5 @@ projectPomDescription = "This module provides a $project.msftAppInsights appende
 // endregion Publishing properties
 
 dependencies {
-    compileOnly group: 'log4j', name: 'log4j', version: '1.2.17'
+    testImplementation group: 'log4j', name: 'log4j', version: '1.2.17'
```
Comment (trask) [inline logging/log4j1_2/build.gradle:36]: you probably need both, see #2002

**Comment by heyams** [comment]
@trask it didn't work it seems. 

Included in Log4j 1.2 is a SocketServer class that is vulnerable to deserialization of untrusted data which can be exploited to remotely execute arbitrary code when combined with a deserialization gadget when listening to untrusted network traffic for log data. This affects Log4j versions up to 1.2 up to 1.2.17.

Users are advised to migrate to org.apache.logging.log4j:log4j-core

should we drop 1.2 entirely? there is no log4j-core:1.x. or should we dismiss it manually?

**Comment by trask** [comment]
what exactly doesn't work?

with `compileOnly` it shouldn't pick up the vulnerability since you fixed dependency locks to only generate for runtimeClasspath now

**Comment by heyams** [comment]
i meant the test failed.

**Comment by trask** [comment]
the test failure looks related to the xstream upgrade:

```
java.lang.NoClassDefFoundError: com/microsoft/applicationinsights/core/dependencies/xmlpull/mxp1/MXParser
     	at com.microsoft.applicationinsights.core.dependencies.xstream.io.xml.Xpp3Driver.createParser(Xpp3Driver.java:52)
```

**Comment by heyams** [comment]
https://jcenter.bintray.com/eu/infomas/annotation-detector/3.0.5/annotation-detector-3.0.5.pom is not still not found.

**Pull request merged**